openmrs/src/test/java/org/motechproject/openmrs/it/MRSTasksIntegrationBundleIT.java

```diff
@@ -0,0 +1,471 @@
+        tempEncounterType.setDescription("TestDescription");
+
+        String encounterUuid = encounterService.createEncounterType(DEFAULT_CONFIG_NAME, tempEncounterType).getUuid();
+        return encounterService.getEncounterTypeByUuid(DEFAULT_CONFIG_NAME, encounterUuid);
+    }
+
+    private boolean waitForTaskExecution(Long taskID) throws InterruptedException {
+        getLogger().info("testOpenMRSTasksIntegration starts waiting for task to execute");
+        int retries = 0;
+        while (retries < MAX_RETRIES_BEFORE_FAIL && !hasTaskExecuted(taskID)) {
+            retries++;
+            Thread.sleep(WAIT_TIME);
+        }
+        if (retries == MAX_RETRIES_BEFORE_FAIL) {
+            getLogger().info("Task execution failed");
+            return false;
+        }
+        getLogger().info("Task executed after " + retries + " retries, what took about "
+                + (retries * WAIT_TIME) / 1000 + " seconds");
+        return true;
+    }
+
+    private boolean hasTaskExecuted(Long taskID) {
+        Set<TaskActivityType> activityTypes = new HashSet<>();
+        activityTypes.add(TaskActivityType.SUCCESS);
+        QueryParams queryParams = new QueryParams((Order) null);
+        List<TaskActivity> taskActivities = taskActivitiesDataService.byTaskAndActivityTypes(taskID, activityTypes, queryParams);
+
+        return taskActivities.size() == 1 ? true : false;
+    }
+
+    private DataSource createPatientDataSource() {
+        List<Lookup> lookupList = new ArrayList<>();
+        lookupList.add(new Lookup("openMRS.uuid", createdPatient.getUuid()));
+        DataSource dataSource = new DataSource(OPENMRS_MODULE_NAME, new Long(4), new Long(0), "Patient-" + DEFAULT_CONFIG_NAME, "openMRS.lookup.uuid", lookupList, false);
+        dataSource.setOrder(0);
+        return dataSource;
+    }
+
+    private DataSource createEncounterDataSource() {
+        List<Lookup> lookupList = new ArrayList<>();
+        lookupList.add(new Lookup("openMRS.uuid", createdEncounter.getUuid()));
+        DataSource dataSource = new DataSource(OPENMRS_MODULE_NAME, new Long(4), new Long(0), "Encounter-" + DEFAULT_CONFIG_NAME, "openMRS.lookup.uuid", lookupList, false);
+        dataSource.setOrder(0);
+        return dataSource;
+    }
+
+    private DataSource createProviderDataSource() {
+        List<Lookup> lookupList = new ArrayList<>();
+        lookupList.add(new Lookup("openMRS.uuid", createdProvider.getUuid()));
+        DataSource dataSource = new DataSource(OPENMRS_MODULE_NAME, new Long(4), new Long(0), "Provider-" + DEFAULT_CONFIG_NAME, "openMRS.lookup.uuid", lookupList, false);
+        dataSource.setOrder(0);
+        return dataSource;
+    }
+
+    private void checkIfPatientWasCreatedProperly() {
+        Patient patient = patientService.getPatientByMotechId(DEFAULT_CONFIG_NAME, String.format("%staskCreated", MOTECH_ID));
+
+        assertEquals(createdPatient.getUuid(), patient.getPerson().getPreferredAddress().getAddress2());
```

Comment:
I've noticed that some values are not checked, for example patient.display here, is there any reason for that ? I think it would be better to verify all used values